sound/soc/soc-pcm.c

```diff
@@ -940,6 +939,23 @@ static int soc_pcm_hw_params(struct snd_pcm_substream *substream,
 	if (ret < 0)
 		goto out;
 
+	for_each_rtd_cpu_dais(rtd, i, cpu_dai) {
+		/*
+		 * Skip CPUs which don't support the current stream
+		 * type. See soc_pcm_init_runtime_hw() for more details
+		 */
+		if (!snd_soc_dai_stream_valid(cpu_dai, substream->stream))
+			continue;
+
+		ret = snd_soc_dai_hw_params(cpu_dai, substream, params);
```

Comment:
snd_soc_dai_hw_params will call snd_soc_link_be_hw_params_fixup to change the params.
In original sequence,  codec_params will use default params. With this commit, it will use the params fixed up by cpu_dai. This may be a risk.

" /* copy params for each codec */
		codec_params = *params;
"

Reply:
I removed the order change in my last version, not worth the effort.